src/Refract.purs

```diff
@@ -366,8 +408,10 @@ run elemId cmp st updateState = void $ element >>= render ui
     spec :: R.ReactSpec Unit st (ReactEff eff)
     spec = R.spec st \this -> do
       st' <- R.readState this
-      _   <- unsafeCoerce updateState $ st'
-      pure $ cmp (\effect -> unsafeCoerce $ launchAff_ $ interpretEffect this (unsafeCoerce effect)) st'
+      unsafeCoerceEff $ updateState $ st'
+      pure $ st # cmp \effect ->
```

Comment:
Small mistake here - should be `st'` and not `st`, `st` refers to the initial state, so no effect ever takes place :)